### PR TITLE
Improve CustomerValidationFactory

### PR DESCRIPTION
**Pull request by fpesch**
### 1. Why is this change necessary?
It improves the codes performance and readability.

### 2. What does this change do, exactly?
* Changes order of method calls
* Replaces merge process with less bloated and more performant approach

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Review by philipreinken — CHANGES_REQUESTED** [review]
Hi @bpesch , thank you for creating this pull request 🙂 Please add a test for your change, so we can be sure this issue won't come up again, and the code works as intended.

File: changelog/_unreleased/2021-03-24-improve-customer-validation-factory.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+title: Improve-customer-validation-factory   
+issue: NONE   
+author: bpesch   
+author_email: b.pesch@protonmail.com  
+author_github: bpesch
+---
+
+# Core
+
+* Improved ````CustomerValidationFactory::update```` performance
```
Comment (philipreinken) [inline changelog/_unreleased/2021-03-24-improve-customer-validation-factory.md]:
```suggestion
* Changed `CustomerValidationFactory::update` to improve performance
```

Comment (philipreinken) [reply changelog/_unreleased/2021-03-24-improve-customer-validation-factory.md]: Please change the other line accordingly

**Review by philipreinken — CHANGES_REQUESTED** [review]

**Comment by fpesch** [comment]
Hi @philipgatzka,
the `CustomerValidationFactory` does not seem to have tests written for the `update` and `create` methods yet.
Besides that, those improvements are performance improvements and should not change the codes functionality at all.
Changelog was updated.
Best regards!

**Review by philipreinken — CHANGES_REQUESTED** [review]
Hi @bpesch , you're right, the change should not cause a difference in the output, but we'd like to make sure that doesn't happen (currently as well as in future commits). You don't need to implement a test case for the whole class, but only one making sure the `$definition` returned by `create()` is altered as expected. If you need any assistance implementing this with PHPUnit, please feel free to ask 🙂

**Comment by fpesch** [comment]
Hi @philipgatzka,
I didn't have a lot of time in the past weeks. Sorry for my late answer.
Please let me know if the added test case is sufficient!

**Review by philipreinken — APPROVED** [review]
Hi @bpesch , no worries! Thank you for adding the test. Looks great 🙂

**Comment by fpesch** [comment]
Should I add the ticket id to the changelog file?

**Comment by philipreinken** [comment]
No, it's fine. Since we're creating the ticket in the process, we usually add another commit on top to add the issue key.

**Comment by philipreinken** [comment]
Thank you for your contribution @bpesch! 🚀

**Pull request closed without merging**